internal/pkg/util/yamlprotoconverter_test.go

```diff
@@ -0,0 +1,271 @@
+package yamlprotoconverter_test
+
+import (
+	. "github.com/envoyproxy/xds-relay/internal/pkg/util"
+	aggregationv1 "github.com/envoyproxy/xds-relay/pkg/api/aggregation/v1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	"google.golang.org/protobuf/proto"
+)
+
+type AndResult = aggregationv1.ResultPredicate_AndResult
+type Fragment = aggregationv1.KeyerConfiguration_Fragment
+type FragmentRule = aggregationv1.KeyerConfiguration_Fragment_Rule
+type KeyerConfiguration = aggregationv1.KeyerConfiguration
+type MatchPredicate = aggregationv1.MatchPredicate
+type RegexAction = aggregationv1.ResultPredicate_ResultAction_RegexAction
+type RequestNodeFragment = aggregationv1.ResultPredicate_RequestNodeFragment
+type RequestTypeMatch = aggregationv1.MatchPredicate_RequestTypeMatch
+type ResourceNamesFragment = aggregationv1.ResultPredicate_ResourceNamesFragment
+type ResultAction = aggregationv1.ResultPredicate_ResultAction
+type ResultPredicate = aggregationv1.ResultPredicate
+type StringFragment = aggregationv1.ResultPredicate_StringFragment
+
+var positiveTests = []TableEntry{
+	{
+		Description: "test result predicate containing string_fragment",
+		Parameters: []interface{}{
+			`
+string_fragment: abc
+`,
+			&ResultPredicate{
+				Type: &StringFragment{
+					StringFragment: "abc",
+				},
+			},
+		},
+	},
+	{
+		Description: "test result predicate containing resource_names_fragment",
+		Parameters: []interface{}{
+			`
+resource_names_fragment:
```

Comment:
I think it would be cleaner to put these yaml definitions in a `testdata` fixtures directory,
and rather than `[]interface{}` make `Parameters` type
```
[]struct {
    yamlPath string
    proto []interface{}
}
```
(or even making `proto` more strict in the above example to `ResultPredicate` and `KeyerConfiguration`

Reply:
@jessicayuen , we cannot "upcast" to a [slice of interface](https://golang.org/doc/faq#convert_slice_of_interface).